core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -1143,14 +1156,20 @@ public static Expression convert(Expression operand, Type fromType,
                 "toString",
                 operand));
       } else {
-        // E.g. from "BigDecimal" to "String"
-        // Generate "x == null ? null : x.toString()"
-        return Expressions.condition(
-            Expressions.equal(operand, RexImpTable.NULL_EXPR),
-            RexImpTable.NULL_EXPR,
-            Expressions.call(
-                operand,
-                "toString"));
+        Expression result;
+        try {
+          // Try to call "toString()" method
```

Comment:
Instead of try ... catch ... clause, can we use the java reflection to detect if the class has "toString" method first ?

Reply:
Do you have any advices on efficient code:)? I put some thought to this problem, but failed to get an elegant way, it seems that we should enumerate all methods to check their method names, argument types and return type.
Meanwhile, I found that in RexToLixTranslator, there is already a static function to find method in a class through reflection. But we still need to catch exception. `Expressions.call` also adopts the similar implementation.

```
  private static Method findMethod(
      Class<?> clazz, String name, Class... parameterTypes) {
    try {
      return clazz.getMethod(name, parameterTypes);
    } catch (NoSuchMethodException e) {
      throw new RuntimeException(e);
    }
  }
```